### PR TITLE
Updated namespace with keyInfo item; added 'anyOption' to records

### PR DESCRIPTION
**Pull request by hudaif747**
### Checklist

**I, the author of this PR checked the following requirements for good software quality:**

- [ ] The code is properly formatted (I ran the formatter)
- [ ] The code is written with our software quality standards (I ran the linter)
- [ ] The code is written using our code style
- [ ] Extensive in source documentation has been added
- [ ] Unit and/or integration tests have been added
- [ ] All texts have been internationalized with at least the following languages:
  - [ ] English
  - [ ] German
- [ ] I tried addressing all new accessibility problems displayed in the console and documented if they can't be fixed
- [ ] I attached performance measurements to prevent performance degradation
- [ ] I added the changes to the next release section of the [changelog](../frontend/docs/changelog/changelog-en.md)

**I, the reviewer checked the following things:**

- [ ] I ran the software once and tried all new and related functionality to this PR
- [ ] I looked at all new and changed lines of code and commented on possible problems
- [ ] I read the added documentation and checked if it is understandable and clear
- [ ] I checked the added tests for completeness
- [ ] I checked the internationalized strings for spelling errors
- [ ] I checked the performance metrics for problems or unexplained degradation
- [ ] I checked that the changes are noted in the [changelog](../frontend/docs/changelog/changelog-en.md)

### CODE REVIEW

File: frontend/src/types/pandemos.ts [hunk 1]
```diff
@@ -3,31 +3,37 @@
 
 /** Display names for the diffent enums of the pandemos data */
 export namespace KeyInfo {
+  export const anyOption: KeyInfoItem = {
+    icon: '🤷',
+    fullName: "Any/Don't Care",
```
Comment (NXXR) [inline frontend/src/types/pandemos.ts:8]:
```suggestion
    fullName: "Any",
```

File: frontend/src/types/pandemos.ts [hunk 2]
```diff
@@ -3,31 +3,37 @@
...
-    /**     Cemetery */ 10: '⚰',
+  export const location_type: Record<number, KeyInfoItem> = {
+    /**          Any */ 0: anyOption,
+    /**         Home */ 1: {icon: '🏡', fullName: 'Home'},
+    /**       School */ 2: {icon: '🏫', fullName: 'School'},
+    /**         Work */ 3: {icon: '🏭/🏢', fullName: 'Workplace'},
+    /** Social Event */ 4: {icon: '🏟', fullName: 'Social Event'},
+    /**     Shopping */ 5: {icon: '🏪', fullName: 'Shopping'},
+    /**     Hospital */ 6: {icon: '🏥❗', fullName: 'Hospital'},
+    /**          ICU */ 7: {icon: '🏥‼', fullName: 'Intensive Care Unit'},
```
Comment (NXXR) [inline frontend/src/types/pandemos.ts:19]:
```suggestion
    /**          ICU */ 7: {icon: '🏥‼', fullName: 'Intensive Care'},
```

File: frontend/src/types/pandemos.ts [hunk 3]
```diff
@@ -3,31 +3,37 @@
...
-    /**         Unknown */ 6: '❓',
+  export const transport_mode: Record<number, KeyInfoItem> = {
+    /**             Any */ 0: anyOption,
+    /**            Bike */ 1: {icon: '🚴‍♀️', fullName: 'Bicycle'},
+    /**    Car (Driver) */ 2: {icon: '🚘👤', fullName: 'Car as Driver'},
+    /** Car (Passenger) */ 3: {icon: '🚘👥', fullName: 'Car as Passenger'},
+    /**             Bus */ 4: {icon: '🚍', fullName: 'Bus'},
+    /**         Walking */ 5: {icon: '🚶‍♀️', fullName: 'Walking'},
+    /**           Other */ 6: {icon: '🛸', fullName: 'Other Mode'},
+    /**         Unknown */ 7: {icon: '❓', fullName: 'Unknown Mode'},
```
Comment (NXXR) [inline frontend/src/types/pandemos.ts:34]:
```suggestion
    /**           Other */ 6: {icon: '🛸', fullName: 'Other'},
    /**         Unknown */ 7: {icon: '❓', fullName: 'Unknown'},
```

File: frontend/src/types/pandemos.ts [hunk 4]
```diff
@@ -38,16 +44,17 @@ export namespace KeyInfo {
...
+  export const infection_state: Record<number, KeyInfoItem> = {
+    /**                             Any */ 0: anyOption,
+    /**                     Susceptible */ 1: {icon: '🙂', fullName: 'Susceptible to Infection'},
+    /**       Infected with no symptoms */ 2: {icon: '🤔', fullName: 'Asymptomatic Infection'},
+    /**          Infected with symptoms */ 3: {icon: '🤧', fullName: 'Symptomatic Infection'},
+    /**   Infected with severe symptoms */ 4: {icon: '🤒', fullName: 'Severely Symptomatic'},
+    /** Infected with critical symptoms */ 5: {icon: '🤮', fullName: 'Critically Symptomatic'},
+    /**                       Recovered */ 6: {icon: '😀', fullName: 'Recovered from Infection'},
+    /**                            Dead */ 7: {icon: '💀', fullName: 'Deceased'},
+    /**                         Unknown */ 8: {icon: '❓', fullName: 'Unknown State'},
```
Comment (NXXR) [inline frontend/src/types/pandemos.ts:56]:
```suggestion
    /**                         Unknown */ 8: {icon: '❓', fullName: 'Unknown'},
```

**Review by NXXR — CHANGES_REQUESTED** [review]
👍 
the Prettier Job failed check if running `npm run format` makes things better or worse.
if worse you can either remove the leading comments in the maps since I assume they are the reason why the formatter is unhappy. Otherwise, you can probably also exclude this file from the formatter somehow.

**Comment by NXXR** [comment]
I also checked the differences between `feature/pandemos-datasocket-preprocessing` and `feature/pandemos` there are barely any changes.  
The files were moved and the `KeyInfo`-Stuff was moved from the datasocket to the types file.

Other than that the only major change is that the raw data is served as an array not a crossfilter anymore, and therefore an additional large crossfilter object with the agent and location infos merged into the trips is provided for @Pawandeep-Kaur-Betz statistics and @apoorvakay might also use this object to filter and display the map (if she doesn't use the raw trip and location array and filter based on the `PandemosFilterSlice` in the store)

You can check the datasocket code if there are any comments or todos that need fixing but I think most of it was optional for now.  
If all looks good you can also merge the `feature/pandemos-datasocket-preprocessing` branch into the `feature/pandemos` branch.  
make sure to let the others know as they probably need to pull the changes into their branches and modify their code accordingly.

**Review by NXXR — APPROVED** [review]

**Comment by NXXR** [comment]
Did you find the issue with the formatter @hudaif747 ?

**Comment by hudaif747** [comment]
> Did you find the issue with the formatter @hudaif747 ?

The formatter fixed single qoutes in the pandemos.ts it did not mess with the rest of the file. The lint fix suggested to add disable comment for namespace and also to remove non-null assertions in the PandemosContext.tsx

**Pull request merged**